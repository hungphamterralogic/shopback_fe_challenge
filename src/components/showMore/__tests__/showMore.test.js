import React from 'react';
import renderer from 'react-test-renderer';
import ShowMore from '../index';
import detectDevice from '../../../utils/detectDevice';

describe('component/showMore', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<ShowMore />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('is Mobile', () => {
    jest.spyOn(detectDevice, 'isMobile').mockImplementationOnce(() => true);

    const tree = renderer.create(<ShowMore />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('is not Mobile', () => {
    jest.spyOn(detectDevice, 'isMobile').mockImplementationOnce(() => false);

    const tree = renderer.create(<ShowMore />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
