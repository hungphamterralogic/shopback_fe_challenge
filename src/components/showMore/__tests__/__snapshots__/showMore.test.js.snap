// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/showMore is Mobile 1`] = `
<div
  className="ShowMore-showMoreMobile-0-1-2"
>
  <div
    className="ShowMore-textBoxMobile-0-1-4"
  >
    All Stores
  </div>
</div>
`;

exports[`component/showMore is not Mobile 1`] = `
<div
  className="ShowMore-showMoreDesktop-0-1-1"
>
  <div
    className="ShowMore-textBoxDesktop-0-1-3"
  >
    view all popular stores
  </div>
</div>
`;

exports[`component/showMore renders correctly 1`] = `
<div
  className="ShowMore-showMoreDesktop-0-1-1"
>
  <div
    className="ShowMore-textBoxDesktop-0-1-3"
  >
    view all popular stores
  </div>
</div>
`;
